snapcraft_legacy/internal/repo/apt_cache.py

```diff
@@ -120,7 +120,15 @@ def _populate_stage_cache_dir(self) -> None:
 
         # Copy current cache configuration.
         cache_etc_apt_path.parent.mkdir(parents=True, exist_ok=True)
-        shutil.copytree("/etc/apt", cache_etc_apt_path)
+
+        # systems with ubuntu pro have an auth file in /etc/apt readable only by root
+        # this may raise an error when copying /etc/apt into the cache directory
```

Comment:
Code looks ok, and approving it, but this comment needs to be improved: it's not `/etc/apt` that is not readable; the directory itself is readable just fine (otherwise `copytree` will fail differently) but the files inside are only readable by root).

Furthermore, to think about: do we want to support other errors here? (e.g. `/etc/apt` being non readable, which will fail with a "standard" PermissionError).

Reply:
- [X] improved the comments
- [X] caught and tested `PermissionError` when `/etc/apt` is unreadable